util/util.py

```diff
@@ -119,9 +119,36 @@ def get_challenge_by_name(database, challenge_name, ctf_channel_id):
     return None
 
 
+def get_challenge_from_args_or_channel(database, args, channel_id):
+    """
+    Helper method for getting a Challenge either from arguments or current channel.
+    Return the corresponding Challenge if called from a challenge channel.
+    Return the Challenge correpsonding to the first argument if called from the
+    CTF channel.
+    Return None if no Challenge can be found.
+    """
+
+    # Check if we're currently in a challenge channel
+    current_chal = get_challenge_by_channel_id(database, channel_id)
+
+    if current_chal:
+        # User is in the challenge channel
+        challenge = current_chal
+    else:
+        # Assume user is in the ctf channel
+        try:
+            challenge_name = args[0].lower().strip("*")
```

Comment:
I don't like bits like this copied around. Maybe we should have a 'clean' utility that takes out special characters and does `.lower`?

Reply:
I agree but think it should be part of a different PR. I have created an issue for it https://github.com/OpenToAllCTF/OTA-Challenge-Bot/issues/227